### PR TITLE
feat(query): Introduce custom allocator for HashTable

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #issue

### CODE REVIEW

**Review by zhang2014 — APPROVED** [review]

File: src/common/base/Cargo.toml [hunk 1]
```diff
@@ -30,7 +29,7 @@ common-exception = { path = "../exception" }
 async-trait = "0.1.56"
 ctrlc = { version = "3.2.2", features = ["termination"] }
 futures = "0.3.21"
-libc = { version = "0.2.126", optional = true }
+libc = { version = "0.2.126" }
```
Comment (Xuanwo) [inline src/common/base/Cargo.toml:32]: I guess we need a toml format.

File: src/common/base/src/mem_allocator/mmap_allocator.rs [hunk 2]
```diff
@@ -0,0 +1,304 @@
+// Copyright 2021 Datafuse Labs.
```
Comment (Xuanwo) [inline src/common/base/src/mem_allocator/mmap_allocator.rs:1]:
```suggestion
// Copyright 2022 Datafuse Labs.
```

File: src/common/base/src/mem_allocator/stackful_allocator.rs [hunk 3]
```diff
@@ -0,0 +1,174 @@
+// Copyright 2021 Datafuse Labs.
```
Comment (Xuanwo) [inline src/common/base/src/mem_allocator/stackful_allocator.rs:1]: ditto

File: src/common/base/src/mem_allocator/mod.rs [hunk 4]
```diff
@@ -12,12 +12,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-mod allocators;
+mod je_allocator;
```
Comment (Xuanwo) [inline src/common/base/src/mem_allocator/mod.rs:15]: Maybe we can add a new crate called `common-allocator` in the following PRs?

Cc @PsiACE

**Comment by Xuanwo** [comment]
This PR will break build on macOS.

File: src/common/base/src/mem_allocator/mmap_allocator.rs (hunk 2 above)
Comment (sundy-li) [reply src/common/base/src/mem_allocator/mmap_allocator.rs:1]: Do we need another pr to update all the copyrights?

Comment (Xuanwo) [reply src/common/base/src/mem_allocator/mmap_allocator.rs:1]: Up to you. Both LGTM.

Comment (BohuTANG) [reply src/common/base/src/mem_allocator/mmap_allocator.rs:1]: > Do we need another pr to update all the copyrights?

No.
The year of the copyright means the file which year was created.

**Review by Xuanwo — APPROVED** [review]

File: src/common/base/src/mem_allocator/mod.rs (hunk 4 above)
Comment (Xuanwo) [reply src/common/base/src/mem_allocator/mod.rs:15]: Tracked at https://github.com/datafuselabs/databend/issues/7223

**Pull request merged**